File: IT/Containers/Esercizi/C++/esercizio30.cpp
/*
file esercizio30.cpp
compilare con: g++ -std=c++11 esercizio30.cpp
*/

#include <iostream>
#include <unordered_map>

using std::cin;
using std::cout;
using std::endl;
using std::string;
using std::unordered_map;

int main()
{
	unordered_map<string, string> messages = {{"it", "Benvenuto!"}, {"en", "Welcome!"}, {"en-us", "Welcome!"}};

	cout << "Seleziona la tua lingua: ";

	string lang;
	getline(cin, lang);

	if (messages.find(lang) != messages.end()) {
		cout << messages[lang] << endl;
	}
	else {
		if (lang.empty()) {
			cout << messages["en"] << endl;
			return 0;
		}

		cout << "Lingua sconosciuta!" << endl << "Aggiungere nuova lingua? s/n: ";
		string s;
		getline(cin, s);

		if (s == "s") {
			cout << "Nuovo messaggio di benvenuto: ";
			string msg;
			getline(cin, msg);
			messages[lang] = msg;

			cout << messages[lang] << endl;

			cout << "Messaggi di benvenuto aggiornati." << endl;

			for (auto& msg : messages) {
				cout << msg.first << "=" << msg.second << endl;
			}
		}
	}
}

File: IT/1/Containers/Esercizi/Java/Esercizio30.java
/* file Esercizio30.java */

import java.io.Console;
import java.util.Map;
import java.util.Map.Entry;
import java.util.HashMap;

public class Esercizio30 {
	public static void main(String [] args) {
		Map<String, String> messages = new HashMap<String, String>();

		messages.put("it", "Benvenuto!");
		messages.put("en", "Welcome!");
		messages.put("en-us", "Welcome!");

		String lang = System.console().readLine("Seleziona la tua lingua: ");

		if (messages.containsKey(lang)) {
			System.out.println(messages.get(lang));
		}
		else {
			if (lang.isEmpty()) {
				System.out.println(messages.get("en"));
				return;
			}

			Console console = System.console();
			String s = console.readLine("Lingua sconosciuta!" + System.lineSeparator() + "Aggiungere una nuova lingua? s/n: ");

			if (s.equalsIgnoreCase("s")) {
				String msg = console.readLine("Nuovo messaggio di benvenuto: ");
				messages.put(lang, msg);
				System.out.println(messages.get(lang));

				System.out.println("Messaggi di benvenuto aggiornati.");

				for (Entry<String, String> entry : messages.entrySet()) {
					System.out.println(entry.getKey() + "=" + entry.getValue());
				}
			}
		}
	}
}

